src/Symfony/Component/FeatureToggle/StrategyResult.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\FeatureToggle;
+
+enum StrategyResult
+{
+    case Grant;
+    case Deny;
+    case Abstain;
+
+    public function isEnabled(bool $fallback): bool
```

Comment:
I don't think this should be a method on the StrategyResult` as this would encourage strategies to rely on that method while they should not. As this is only needed inside `Feature::isEnabled` AFAICT, I would put the logic there instead.

Reply:
It could be both. To me it makes sense to keep it in StrategyResult as it is the one holding the Grant/Deny decision to convert it to a bool value. Let me know. and Strategies don't rely on this method because Feature::isEnabled already calls this method.

Reply:
Well, that's precisely my point. The only place where it is **legitimate** to use this method is inside `Feature::isEnabled`, which is also in the component. So there is no reason to move this logic in the enum. It only encourages authors of custom strategies to use it while they should never use it.

Reply:
moved